Assets/LeapMotion/Core/Scripts/Attributes/ImplementsInterface.cs

```diff
@@ -55,35 +57,78 @@ public class ImplementsInterfaceAttribute : CombinablePropertyAttribute,
     /// that implements that interface, or null if none was found.
     /// </summary>
-    public UnityObject FindImplementer(UnityObject obj) {
-
-      if (!fieldInfo.FieldType.IsAssignableFrom(obj.GetType())
-          && !(typeof(Component).IsAssignableFrom(fieldInfo.FieldType)
-               && obj.GetType() == typeof(GameObject))) {
-        // Even if the object implements the correct interface, the field isn't
-        // compatible with this object. E.g. A ScriptableObject can't be assigned to a
-        // MonoBehaviour field.
+    public UnityObject FindImplementer(SerializedProperty property, UnityObject fromObj) {
+
+      // Don't use fieldInfo here because it is very convenient for it to be left null
+      // on the CombinablePropertyAttribute when dealing with generic-type situations
+      // where the ImplementsInterface class has to be constructed manually in a custom
+      // editor. (Specific case: StreamConnectorEditor.cs)
+
+      bool isTypeDefinitelyIncompatible;
+      {
+        isTypeDefinitelyIncompatible = !this.type.IsAssignableFrom(fromObj.GetType());
+
         // We have to make an exception when a GameObject is dragged into a field whose
-        // type is a Component; we use GetComponent to satisfy that case.
+        // type is a Component, because it's expected that we can use GetComponent in
+        // that case. (We might still fail later if the component isn't found.)
+        var objIsGameObject = fromObj.GetType() == typeof(GameObject);
+        if (objIsGameObject) {
+          isTypeDefinitelyIncompatible = false;
+        }
+
+        // We also make an exception when a Component is dragged into a field that
+        // doesn't directly satisfy the interface because it might have other Components
+        // on its GameObject that _do_ satisfy the field, again via GetComponent.
+        var objIsComponent = typeof(Component).IsAssignableFrom(fromObj.GetType());
+        if (objIsComponent) {
+          isTypeDefinitelyIncompatible = false;
+        }
+
+        // However, you can't assign a ScriptableObject to a field expecting a Component,
+        // or vice-versa, no matter what.
+
+        // Matches the "Foo" in "PPtr<$Foo>" (as an inner group)
+        var fieldTypeMatch = Regex.Match(property.type, "PPtr<\\$(.+)>");
+        if (fieldTypeMatch.Groups.Count == 0) {
+          // No captured groups means we have no idea what the enclosing field type is.
+          isTypeDefinitelyIncompatible = true;
+        }
+        else {
+          // First group is the whole match, second is the inner group, the class name.
+          var fieldTypeName = fieldTypeMatch.Groups[1].Value;
+
+          var fieldIsScriptableObject = fieldTypeName.Equals("ScriptableObject");
+          if (fieldIsScriptableObject && (objIsComponent || objIsGameObject)) {
+            isTypeDefinitelyIncompatible = true;
+          }
+
+          var fieldTakesComponent = fieldTypeName.Equals("MonoBehaviour")
```

Comment:
There are more ways for a field to accept a Component.  Really, the field could be anything that inherits from Component.  Behaviour is one that exists inside the UnityEngine namespace even, which is just a component that can enabled or disabled.

There is a lot of logic added to allow fieldInfo to be passed in as null, might it be easier to get your hands on the fieldInfo in question?

Reply:
>There is a lot of logic added to allow fieldInfo to be passed in as null, might it be easier to get your hands on the fieldInfo in question?

Yes it would! I was unable to actually get access to it. I made an attempt the first time to try and avoid having to get rid of passing it in from the custom property, then I tried a second time when I was addressing your earlier comment on this method in the PR. I just couldn't figure out how to get it.. Maybe I was just missing something simple?

Reply:
If this is inside a PropertyDrawer, you should already have access to the fieldInfo directly.  If this is inside a custom editor, you would have to get the fieldInfo manually with reflection like `target.GetType().GetField(property.name)`.